tests/test_cli.py

```diff
@@ -19,10 +20,39 @@ def test_cli_version():
 def test_schema():
     with tempfile.TemporaryDirectory() as tmp:
         tmp_path = Path(tmp)
+
         # Generate up to date schema files
         run_cli("schema", tmp_path)
+
         # Compare with the expected schema files
         for schema in tmp_path.glob("*.json"):
             expected = PATH_TO_SCHEMAS / schema.name
             if schema.read_text() != expected.read_text():
                 raise AssertionError(f"Schema file {expected} is out of date.")
+
+
+def test_demo_configuration(samples: Path, demo_config: Path):
+    # use a fixed path for the demo configuration so that the outputs are consistent
+    out_folder = "deploy-tools-output"
+    temp_out = Path("/tmp") / out_folder
+    expected_out = samples / out_folder
+
+    # make sure the output directory is empty and exists
+    rmtree(temp_out, ignore_errors=True)
+    temp_out.mkdir(exist_ok=True)
+
+    # generate the demo configuration output
+    run_cli("sync", "--from-scratch", str(temp_out), str(demo_config))
+
+    # compare the output with the expected output
+    for expected in expected_out.glob("**/*"):
```

Comment:
It seems that running generate_samples is a bit confusing to new contributors, in the same way that generate_schema was.

Reply:
Fair enough. This needs documenting. But I stand by the approach. When you have made changes you run it and then look at the diff and make sure you are happy with it before committing the new samples.

Reply:
But perhaps it could be added as a Python script instead of a shell script? What about adding it as a VSCode task somehow?

I'm not aware of something like `[project.dev-scripts]` in pyproject.toml though...

Reply:
I figure we could add it as a vscode task that calls the script. It's a rather bashy set of steps really.

Reply:
Cool, that sounds sensible

Reply:
I've added the task